File: scat/src/main/java/scat/domain/model/School.java
package scat.domain.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

/**
 * @author levry
 */
@Data
@Entity
@EqualsAndHashCode(of = "id")
public class School {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull
    @Column(nullable = false)
    private String name;

    @Column
    private Integer number;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "type")
    private SchoolType type;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "city")
    private City city;

}

File: scat-test/src/main/kotlin/scat/data/School.kt
package scat.data

data class School (
        val id: Int,
        val name: String,
        val number: Int?,
        val type: Entity,
        val country: Entity,
        val region: Entity?,
        val city: Entity
)

data class SchoolData (
        val name: String,
        val number: Int?,
        val type: Int,
        val city: Int
)